localstack/services/awslambda/lambda_utils.py

```diff
@@ -185,9 +186,11 @@ def store_lambda_logs(
 
 
 def get_main_endpoint_from_container() -> str:
-    global DOCKER_MAIN_CONTAINER_IP
     if config.HOSTNAME_FROM_LAMBDA:
         return config.HOSTNAME_FROM_LAMBDA
+    if config.LAMBDA_EXECUTOR == "local":
+        # special case: return localhost for local Lambda executor (TODO remove after full switch to v2 provider)
+        return LOCALHOST
```

Comment:
I am not a fan of this, as it impacts the v2 provider even though it shouldn't. 
Example: People setting `PROVIDER_OVERRIDE_LAMBDA=v2` + `LAMBDA_EXECUTOR=local`. Does not make sense to set it, but it will effectively brick the new provider, although it should not influence it.
I would at least check which provider is active, or better handle this in the calling method of `get_main_endpoint_from_container` in the old provider :)

Reply:
Great catch @dfangl , will look into that. The main use case was coming from a [new integration test](https://github.com/localstack/localstack/pull/7905/files#diff-03f6635c4fa95c5cddc4edb72266f76f9962986897cb93eb0293f687022a7e02R798) which needs to pass the LocalStack endpoint to a Lambda function (currently, we're still running the tests with the `local` executor of the old provider). 

Afaict, we currently don't yet have a lookup mechanism that would determine the endpoint for all combinations of `PROVIDER_OVERRIDE_LAMBDA`/`LAMBDA_EXECUTOR` - but as you mentioned, we could probably control this from the calling method (integration test function) for the time being. 👍

Reply:
Done @dfangl , removed the switch in `lambda_utils.py`, and added it directly to the test function itself.